R/zchunk_L232.industry_CHINA.R

```diff
@@ -0,0 +1,537 @@
+      ungroup %>% filter(year %in% MODEL_BASE_YEARS) %>%
+      rename(region = province) %>%
+      left_join_keep_first_only(calibrated_techs, by = c("sector", "fuel")) %>%
+      select(-calibration, -secondary.output) %>%
+      rename(stub.technology = technology) ->
+      L232.in_EJ_province_indfeed_F_Yh
+    L232.in_EJ_province_indfeed_F_Yh %>% select(LEVEL2_DATA_NAMES[["StubTechYr"]], "value") %>%
+      left_join_keep_first_only(select(A32.globaltech_eff, subsector, technology, minicam.energy.input),
+                                by = c("subsector", "stub.technology" = "technology")) %>%
+      mutate(calibrated.value = round(value, energy.DIGITS_CALOUTPUT),
+             share.weight.year = year,
+             tech.share.weight = if_else(calibrated.value > 0, 1, 0)) %>%
+      group_by(region, supplysector, subsector, year) %>% mutate(x = sum(calibrated.value)) %>%
+      # ^^ sets up variable (x) for defining subsector shareweight
+      mutate(subs.share.weight = if_else(x > 0, 1, 0)) %>% ungroup %>%
+      select(LEVEL2_DATA_NAMES[["StubTechCalInput"]]) ->
+      L232.StubTechCalInput_indfeed_CHINA  ## OUTPUT
+
+    # get industrial sector calibrated output
+    A32.globaltech_eff %>%
+      complete(nesting(supplysector, subsector, technology, minicam.energy.input, secondary.output),
+               year = c(year, MODEL_BASE_YEARS)) %>%
+      group_by(supplysector, subsector, technology, minicam.energy.input, secondary.output) %>%
+      mutate(value = approx_fun(year, value)) %>% ungroup %>%
+      filter(year %in% MODEL_BASE_YEARS) %>%
+      rename(efficiency = value) %>%
+      mutate(efficiency = round(efficiency, energy.DIGITS_EFFICIENCY)) ->
+      A32.globaltech_eff_interp
+
+    L232.in_EJ_province_indenergy_F_Yh %>%
+      bind_rows(L232.in_EJ_province_indfeed_F_Yh) %>%
+      left_join_keep_first_only(select(A32.globaltech_eff_interp, supplysector, subsector, technology, year, efficiency),
+                                by = c("supplysector", "subsector", "stub.technology" = "technology", "year")) %>%
+      mutate(calOutputValue = round(value * efficiency, energy.DIGITS_CALOUTPUT)) ->
+      L232.out_EJ_province_ind_serv_F_Yh
+      # ^^ service output, by technology, for energy-use and feedstocks
+
+    L232.out_EJ_province_ind_serv_F_Yh %>%
+      group_by(region, year) %>%
+      summarise(calOutputValue = sum(calOutputValue)) %>% ungroup %>%
+      # ^^ aggregate to get output of industrial sector in each region
+      mutate(supplysector = "industry",
+             subsector = "industry",
+             stub.technology = "industry",
+             share.weight.year = year,
+             subs.share.weight = if_else(calOutputValue > 0, 1, 0),
+             tech.share.weight = subs.share.weight) %>%
+      select(LEVEL2_DATA_NAMES[["StubTechProd"]]) ->
+      L232.StubTechProd_industry_CHINA  ## OUTPUT
+
+    # get calibrated output of industrial sector
```

Comment:
This part are creates the stub tech coefficients, not calibrated output. Perhaps make the comment more clear.